src/config/plugin.ts

```diff
@@ -73,6 +74,8 @@ export class Plugin implements IPlugin {
 
   pjson!: PJSON.Plugin
 
+  isESM = false
```

Comment:
Can't we completely hide this in ModuleLoader and just call `ModuleLoader.load()` here instead? The `getPackageType` call in the ModuleLoader code should resolve the correct pjson.type. To potentially reduce file io, you could also pass in pjson.type to the `ModuleLoader.load` method maybe?